File: sueldo empleado java/SalaryCalculator.java
public class SalaryCalculator {

    public static double calculateEmployeeSalary(employee employee, overTime overTime, health healthDeduction, pension pensionDeduction, arl arlDeduction) {
        double baseSalary = employee.getBaseSalary();
        double totalSalary = employee.calculatedSalary();

        if (overTime != null) {
            totalSalary += overTime.calculatedSalary(baseSalary);
        }

        if (healthDeduction != null) {
            // Configure the base salary and percentage in the Health object
            healthDeduction.setBaseSalary(baseSalary);
            healthDeduction.setPercentage(0.04); // Set health percentage (4%)
            totalSalary -= healthDeduction.calculatedDeductions();
        }

        if (pensionDeduction != null) {
            // Configure the base salary and percentage in the Pension object
            pensionDeduction.setBaseSalary(baseSalary);
            pensionDeduction.setPercentage(0.04); // Set pension percentage (4%)
            totalSalary -= pensionDeduction.calculatedDeductions();
        }

        if (arlDeduction != null) {
            // Configure the base salary in the Arl object
            arlDeduction.setBaseSalary(baseSalary);
            totalSalary -= arlDeduction.calculatedDeductions();
        }

        return totalSalary;
    }
}

File: sueldo empleado - copia/SalaryCalculator.php
<?php

class SalaryCalculator {
    public static function calculateEmployeeSalary(Employee $employee, Overtime $overtime = null, Health $healthDeduction = null, Pension $pensionDeduction = null, Arl $arlDeduction = null): float {
        $baseSalary = $employee->getBaseSalary();
        $totalSalary = $employee->calculatedSalary();

        if ($overtime) {
            $totalSalary += $overtime->calculatedSalary($baseSalary);
        }

        if ($healthDeduction) {
            // Configurar el salario base y el porcentaje en el objeto Health
            $healthDeduction->setBaseSalary($baseSalary);
            $healthDeduction->setPercentage(0.04); // Configurar el porcentaje de salud (4%)
            $totalSalary -= $healthDeduction->calculatedDeductions();
        }

        if ($pensionDeduction) {
            // Configurar el salario base y el porcentaje en el objeto Pension
            $pensionDeduction->setBaseSalary($baseSalary);
            $pensionDeduction->setPercentage(0.04); // Configurar el porcentaje de pensión (4%)
            $totalSalary -= $pensionDeduction->calculatedDeductions();
        }

        if ($arlDeduction) {
            // Configurar el salario base en el objeto Arl
            $arlDeduction->setBaseSalary($baseSalary);
            $totalSalary -= $arlDeduction->calculatedDeductions();
        }

        return $totalSalary;
    }
}

?>